scipy/linalg/special_matrices.py

```diff
@@ -603,6 +603,41 @@ def companion(a):
     return c
 
 
+def helmert(n):
+    """
+    Create a Helmert matrix of order `n`.
+
+    This has applications in statistics, compositional or simplicial analysis,
+    and in Aitchison geometry.
+
+    Parameters
+    ----------
+    n : int
+        The size of the array to create.
+
+    Returns
+    -------
+    M : (n, n) ndarray
+        The Helmert matrix.
+
+    Examples
+    --------
+    >>> from scipy.linalg import helmert
+    >>> helmert(5)
+    array([[ 0.4472136 ,  0.4472136 ,  0.4472136 ,  0.4472136 ,  0.4472136 ],
+           [ 0.70710678, -0.70710678,  0.        ,  0.        ,  0.        ],
+           [ 0.40824829,  0.40824829, -0.81649658,  0.        ,  0.        ],
+           [ 0.28867513,  0.28867513,  0.28867513, -0.8660254 ,  0.        ],
+           [ 0.2236068 ,  0.2236068 ,  0.2236068 ,  0.2236068 , -0.89442719]])
+
+    """
+    H = np.tril(np.ones((n, n)), -1) - np.diag(np.arange(n))
+    d = np.arange(n) * np.arange(1, n+1)
+    H[0] = 1
+    d[0] = n
+    return H / np.sqrt(d)[:, np.newaxis]
```

Comment:
(I've no opinions on merging this or not, but...)

Shouldn't this take `dtype` and `order` arguments? Also, I don't know how big these things get in practice, but is it worthwhile optimizing this to save memory:

```python
H = np.tril(np.ones((n, n), dtype=dtype, order=order), -1)
H[np.diag_indices_from(H)] -= np.arange(n)
d = np.arange(n) * np.arange(1, n+1)
H[0] = 1
d[0] = n
H /= np.sqrt(d)[:, np.newaxis]
```